source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  TextInputRenderer.swift
+//  AdaptiveCards
+//
+//  Created by mukuagar on 11/02/21.
+//
+
+import AdaptiveCards_bridge
+import AppKit
+
+class TextInputRenderer: NSObject, BaseCardElementRendererProtocol {
+    static let shared = TextInputRenderer()
+    func render(element: ACSBaseCardElement, with hostConfig: ACSHostConfig, style: ACSContainerStyle, rootView: NSView, parentView: NSView, inputs: [BaseInputHandler]) -> NSView {
+        guard let inputBlock = element as? ACSTextInput else {
+            logError("Element is not of type ACSTextInput")
+            return NSView()
+        }
+        let textView = ACRTextInputView()
+        var attributedPlaceHolder: NSMutableAttributedString
+        var attributedInitialValue: NSMutableAttributedString
+        textView.translatesAutoresizingMaskIntoConstraints = false
+        textView.isBordered = true
+        textView.isEditable = true
+//        textView.delegate = self
+        if let maxLen = inputBlock.getMaxLength() {
+            if Int(truncating: maxLen) > 0 {
+                textView.maxLen = Int(truncating: maxLen)
+            }
+        }
+        if inputBlock.getIsMultiline() {
+            // Disable Horizontal scrolling
+            textView.cell?.isScrollable = false
+            textView.cell?.wraps = true
+            // Makes text go to next line
+            textView.cell?.usesSingleLineMode = false
+            } else {
+            // Makes text remain in 1 line even when Ctrl + return pressed
+            textView.cell?.usesSingleLineMode = true
+            textView.maximumNumberOfLines = 1
+            // Make text scroll horizontally
+            textView.cell?.isScrollable = true
+        }
+        var doesPlaceholderExist = false
+        var doesValueExist = false
+        // Check if placeholder and initial value exist
+        if inputBlock.getPlaceholder() != nil {
+            doesPlaceholderExist = true
+            }
+        if inputBlock.getValue() != nil {
+            doesValueExist = true
+            }
```

Comment:
These can written like,
```suggestion
        let doesPlaceholderExist = inputBlock.getPlaceholder() != nil
        let doesValueExist = inputBlock.getValue() != nil
```